### ISSUE TITLE
download only specific files for #245

### ISSUE THREAD

**Issue by ttodua** [jenkinsci/publish-over-ssh-plugin#329]
_(On top of https://github.com/jenkinsci/publish-over-ssh-plugin/pull/245 PR)_
- at this moment, the added feature compares md5 of the files to determine which were changed. However, the plugin had an additional option (checkbox or so), so if that's checked, plugin will only download the specific files (which are presented as "changed/added/removed" from github push action, from example payload it's visible which files are changed on the last commit:  https://pastebin.com/raw/u8u0B4PB ) . so, at first, the plugin downloaded only those files (instead of whole repo) and then it continue doing what it does (comparing md5 etc...)

**Comment by ttodua** [comment]
@parvit

**Comment by parvit** [comment]
@ttodua If i understand correctly, the plugin should be able to understand if its being triggered by a git push and only take the files in the changeset?

**Comment by ttodua** [comment]
> @ttodua If i understand correctly, the plugin should be able to understand if its being triggered by a git push and only take the files in the changeset?

yes I meant that. if that wont be complex and can be done within my proposal.
 if it's not possible to detect what it is being triggered by,  then just maybe check the last push commit and use that info
